packages/smart-contracts/src/contracts/ERC20EscrowToPay.sol

```diff
@@ -0,0 +1,386 @@
+     * @param paymentReference Reference of the payment related.
+     */
+    event RevertedEmergencyClaim(bytes indexed paymentReference);
+
+    /**
+     * @notice Emitted when _withdraw "to payee" has been executed.
+     * @param tokenAddress Address of the ERC20 token smart contract.
+     * @param to Address to the payment issuer, alias payee.
+     * @param amount Amount transfered.
+     * @param paymentReference Reference of the payment related.
+     * @param feeAmount Set to zero when emited by _withdraw function.
+     * @param feeAddress Set to address(0) when emited by _withdraw function. 
+     */
+    event TransferWithReferenceAndFee(
+        address tokenAddress,
+        address to,
+        uint256 amount,
+        bytes indexed paymentReference,
+        uint256 feeAmount,
+        address feeAddress
+    );
+
+    constructor(address _paymentProxyAddress) {
+        paymentProxy = IERC20FeeProxy(_paymentProxyAddress);
+    }
+
+    /**
+     * @notice receive function reverts and returns the funds to the sender.
+     */ 
+    receive() external payable {
+        revert("not payable receive");
+    }
+
+    /** 
+    * @notice Stores the invoice details, and transfers funds to this Escrow contract.
+    * @param _tokenAddress Address of the ERC20 token smart contract.
+    * @param _to Address to the payment issuer, alias payee.
+    * @param _amount Amount to transfer.
+    * @param _paymentRef Reference of the payment related.
+    * @param _feeAmount Amount of fee to be paid.
+    * @param _feeAddress Address to where the fees will be paid.
+    * @dev Uses modifier IsNotInEscrow.
+    * @dev Uses transferFromWithReferenceAndFee() to transfer funds from the msg.sender, 
+    * into the escrowcontract and pays the _fees to the _feeAdress.
+    * @dev Emits RequestInEscrow(_paymentRef) when the funds are in escrow.
+    */
+    function payEscrow(
+        address _tokenAddress,
+        address _to,
+        uint256 _amount,
+        bytes memory _paymentRef,
+        uint256 _feeAmount,
+        address _feeAddress
+    )   
+        external 
+        IsNotInEscrow(_paymentRef)
+    {   
+        if (_amount == 0 || _feeAmount == 0) revert("Zero Value");
+
+        requestMapping[_paymentRef] = Request(
+            IERC20(_tokenAddress),
+            _to,
+            msg.sender,
+            _amount,
+            0,
+            0,
+            false,
+            false
+        );
+        
+        (bool status, ) = address(paymentProxy).delegatecall(
+        abi.encodeWithSignature(
+        "transferFromWithReferenceAndFee(address,address,uint256,bytes,uint256,address)",
+            _tokenAddress,
+            address(this),
+            _amount,
+            _paymentRef,
+            _feeAmount,
+            _feeAddress
+            )
+        );
+        require(status, "transferFromWithReferenceAndFee failed");
+
+        emit RequestInEscrow(_paymentRef);
```

Comment:
I don't think this event is necessary, the event sent by `transferFromWithReferenceAndFee` looks enough to know we have reached the stage "RequestInEscrow"

Reply:
```suggestion
```